src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -28,10 +44,143 @@ import {CDK_MENU, Menu} from './menu-interface';
+  ) {
+    super();
+
+    this._document = document;
+  }
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+
+    this._configChildSubscriptions();
+  }
+
+  /**
+   * Returns true if this MenuBar, or any open Menu triggered by a MenuItem in this MenuBar,
+   * contains the given HTMLElement.
+   * @param element the element to search for.
+   */
+  _contains(element: HTMLElement) {
+    return (
+      this._elementRef.nativeElement.contains(element) ||
+      this._allItems
+        .filter(menuItem => menuItem.hasMenu() && menuItem._menuTrigger!.isMenuOpen())
+        .some(menuItem => menuItem._menuTrigger!._menuPanel!._menu!._contains(element))
+    );
+  }
+
+  /**
+   * Configure child MenuItem subscriptions and ensure that future child elements are also
+   * subscribed to.
+   */
+  private _configChildSubscriptions() {
+    this._allItems
+      .filter(menuItem => menuItem.hasMenu())
+      .forEach(menuItem => this._subscribeToTriggers(menuItem));
+
+    this._allItems.changes.subscribe((menuItems: QueryList<CdkMenuItem>) => {
+      menuItems
+        .filter(menuItem => menuItem.hasMenu())
+        .forEach(menuItem => this._subscribeToTriggers(menuItem));
+    });
+  }
+
+  /**
+   * Subscribe to the MenuItem open and close event emitters which toggle the subscription to
+   * document click events.
+   * @param menuItem the MenuItem to subscribe to.
+   */
+  private _subscribeToTriggers(menuItem: CdkMenuItem) {
+    menuItem._menuTrigger!.opened.subscribe(() => this._startDocumentListener(menuItem));
+    menuItem._menuTrigger!.closed.subscribe((eventType: CloseEventCause) =>
+      this._stopDocumentListener(eventType)
+    );
+  }
+
+  /**
+   * Start listening to Document click events if there is no document click event listener
+   * registered. The listener will close out the open menu when a click occurs outside the
+   * menu tree.
+   * @param menuItem the MenuItem which opened the menu.
+   */
+  private _startDocumentListener(menuItem: CdkMenuItem) {
+    if (!this._isListeningToDocument()) {
+      merge(
+        fromEvent(this._document, 'click') as Observable<MouseEvent>,
+        fromEvent(this._document, 'touchend') as Observable<TouchEvent>
+      )
+        .pipe(takeUntil(this._destroyDocumentSubscription))
+        .subscribe(event => {
+          // we need to consider whether or not the outside click event occurred while the
+          // element is in the Shadow DOM when fetching the click target.
+          const clickTarget = (this._isShadowRoot() && event.composedPath
+            ? event.composedPath()[0]
+            : event.target) as HTMLElement;
+
+          if (!this._contains(clickTarget)) {
+            menuItem.trigger();
+          }
+        });
+    }
+  }
+
+  /**
+   * Stop the document click listener if one is configured.
+   * @param eventType the type of event which closed the menu.
+   */
+  private _stopDocumentListener(eventType: CloseEventCause) {
```

Comment:
Since this doesn't always stop the document listener, maybe call it '_stopDocumentListenerIfAppropriate' to clarify this.